drivers/sensor/microchip/mtch9010/mtch9010.c

```diff
@@ -0,0 +1,792 @@
+	}
+
+	gpio_pin_interrupt_configure_dt(&config->heartbeatLine, GPIO_INT_EDGE_RISING);
+}
+
+int mtch9010_lock_settings(const struct device *dev)
+{
+	const struct mtch9010_config_t *config = dev->config;
+
+	if (!gpio_is_ready_dt(&config->lockLine)) {
+		LOG_INST_ERR(config->log, "Lock line not ready");
+		return -EIO;
+	}
+
+	LOG_INST_INF(config->log, "Locking MTCH9010");
+
+	/* Set lock line */
+	gpio_pin_configure_dt(&config->lockLine, GPIO_OUTPUT_ACTIVE);
+	return 0;
+}
+
+/* Sensor APIs */
+static const struct sensor_driver_api mtch9010_api_funcs = {
+	.sample_fetch = mtch9010_sample_fetch,
+	.channel_get = mtch9010_channel_get,
+};
+
+/* Device Init Macros */
+#define MTCH9010_OPERATING_MODE_INIT(inst)                                                         \
+	COND_CODE_1(DT_INST_ENUM_HAS_VALUE(inst, mtch9010_operating_mode,\
+		capacitive), (MTCH9010_CONDUCTIVE), (MTCH9010_CAPACITIVE))
+
+#define MTCH9010_SLEEP_TIME_INIT(inst)                                                             \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, mtch9010_sleep_period),\
+		(DT_INST_ENUM_IDX(inst, mtch9010_sleep_period)), (0))
+
+#define MTCH9010_OUTPUT_MODE_INIT(inst)                                                            \
+	COND_CODE_0(DT_INST_PROP_OR(inst, mtch9010_extended_output_enable, false),\
+		(MTCH9010_OUTPUT_FORMAT_CURRENT),\
+		(COND_CODE_1(DT_INST_ENUM_HAS_VALUE(inst,\
+		mtch9010_extended_output_format,delta),\
+		((MTCH9010_OUTPUT_FORMAT_DELTA)),\
+		((COND_CODE_1(DT_INST_ENUM_HAS_VALUE(inst,\
+		mtch9010_extended_output_format, current) \
+		, (MTCH9010_OUTPUT_FORMAT_CURRENT), \
+		(COND_CODE_1(DT_INST_ENUM_HAS_VALUE(inst,\
+		mtch9010_extended_output_format, both_delta_and_current),\
+		(MTCH9010_OUTPUT_FORMAT_BOTH),\
+		(MTCH9010_OUTPUT_FORMAT_MPLAB_DATA_VISUALIZER)\
+		))))))))
+
+#define MTCH9010_REF_MODE_INIT(inst)                                                               \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst,\
+	mtch9010_reference_value),\
+	(MTCH9010_REFERENCE_CUSTOM_VALUE), (MTCH9010_REFERENCE_CURRENT_VALUE))
+
+#define MTCH9010_REF_VAL_INIT(inst)                                                                \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst,\
+	mtch9010_reference_value),\
+	(DT_INST_PROP(inst, mtch9010_reference_value)),\
+	(0))
+
+#define MTCH9010_USER_LABEL(inst) inst
+
+#define MTCH9010_DEFINE(inst)                                                                      \
+	LOG_INSTANCE_REGISTER(mtch9010, MTCH9010_USER_LABEL(inst), CONFIG_MTCH9010_LOG_LEVEL);     \
+	static struct mtch9010_data_t mtch9010_data_##inst = {                                     \
+		.reference = MTCH9010_REF_VAL_INIT(inst),                                          \
+		.threshold = DT_INST_PROP(inst, mtch9010_detect_value),                            \
+		.heartbeatErrorState = false,                                                      \
+		.lastResult = {0, 0}};                                                             \
```

Comment:
lastWake also needs to be initialized to zero to not have issues w/ first sample reading